testing/integration/nginx_unprivileged_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package integration
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/gofrs/uuid/v5"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+	"github.com/elastic/elastic-agent/pkg/control/v2/cproto"
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools"
+)
+
+var nginxStatusModule string = `
+
+server {
+  listen 81;
+  server_name localhost;
+
+  access_log off;
+  allow 127.0.0.1;
+  # deny all;
+
+  location /nginx_status {
+    # Choose your status module
+
+    # freely available with open source NGINX
+    stub_status;
+
+    # for open source NGINX < version 1.7.5
+    # stub_status on;
+
+    # available only with NGINX Plus
+    # status;
+
+    # ensures the version information can be retrieved
+    server_tokens on;
+  }
+}
+
+`
+
+// ExtendedRunner is the main test runner
+type NginxUnprivilegedRunner struct {
+	suite.Suite
+	info                   *define.Info
+	agentFixture           *atesting.Fixture
+	policyID               string
+	policyName             string
+	healthCheckTime        time.Duration
+	healthCheckRefreshTime time.Duration
+}
+
+func TestNginxUnprivileged(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: "fleet",
+		Stack: &define.Stack{},
+		OS: []define.OS{
+			{Type: define.Linux},
+		},
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+	})
+
+	runner := &NginxUnprivilegedRunner{
+		info:                   info,
+		healthCheckTime:        time.Minute * 5,
+		healthCheckRefreshTime: time.Second * 5,
+	}
+
+	suite.Run(t, runner)
+}
+
+func (runner *NginxUnprivilegedRunner) SetupSuite() {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	cmd := exec.CommandContext(ctx, "apt-get", "install", "-v", "nginx-full")
```

Comment:
This is not distro agnostic. You are making this debian specific, so you need to define this in the `define.Require` above.

See above.